clients/da-vinci-client/src/test/java/com/linkedin/davinci/store/rocksdb/RocksDBThrottlerTest.java

```diff
@@ -2,132 +2,330 @@
-
-    public TestOperationRunnable(int threadId, CountDownLatch latch, int waitTimeMs) {
-      this.threadId = threadId;
-      this.latch = latch;
-      this.waitTimeMs = waitTimeMs;
-    }
-
-    @Override
-    public Void execute() throws RocksDBException {
-      LOGGER.info(System.currentTimeMillis() + ", Get function invoked in thread: " + threadId);
-      if (latch != null) {
-        latch.countDown();
-      }
-      Utils.sleep(waitTimeMs);
-      return null;
-    }
-  }
-
-  @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Param: allowedMaxOperationsInParallel should be positive, but.*")
-  public void testThrottleBadInput() {
-    new RocksDBIngestThrottler(0);
-  }
+  private static final Logger LOGGER = LogManager.getLogger(RocksDBThrottlerTest.class);
 
   /**
    * Liveness: Testing whether all the tasks invoking the throttlers gets executed.
-   * - Timeout passed to shutdownExecutor is more than the time it would require to finish all tasks
-   *   resulting in no tasks getting interrupted.
-   * - allowedMaxOperationsInParallel is set to 1, number of tasks is set to 5 and each tasks will
-   *   sleep for 1 sec, so in 5 seconds all the tasks should have finished executing.
-   * - Verified using CountDownLatch to account for the number of tasks getting executed
+   * Correctness: Testing whether only the allowedMaxOperationsInParallel are executed at any time
+   *
+   * Create a throttler with limit 1
+   * Submit a task to thread pool that will call throttledOperation with latch.await() as the payload.
+   * Submit another task to thread pool that will call throttledOperation with empty lambda as the payload.
+   * Assert that the second task is not complete.
+   * Release the latch.
+   * Assert that the first task is complete.
+   * Assert that the second task is complete.
    */
-  @Test(timeOut = 10 * Time.MS_PER_SECOND)
-  public void testThrottleLiveness() throws InterruptedException {
-    int numTasks = 5;
-    int waitTimeMS = 1 * Time.MS_PER_SECOND; // 1 sec
-    int bufferWaitMS = 10 * waitTimeMS; // extra buffer time
-    ExecutorService executorService = Executors.newFixedThreadPool(numTasks);
-    // Create a CountDownLatch with the number of tasks
-    CountDownLatch latch = new CountDownLatch(numTasks);
+
+  @Test(timeOut = 20 * Time.MS_PER_SECOND)
+  public void testThrottleWith2Tasks() throws InterruptedException {
+    int allowedMaxOperationsInParallel = 1;
+    /**
+     * Number of threads is set up to more than the number of tasks to have enough threads
+     * to have the ability to schedule all the tasks at the same time if needed and then
+     * test the throttler's functionality.
+     */
+    ExecutorService executorService = Executors.newFixedThreadPool(5);
+    CountDownLatch taskWaitLatch = new CountDownLatch(1);
+    CountDownLatch task1StartStatusLatch = new CountDownLatch(1);
+    CountDownLatch task2StartStatusLatch = new CountDownLatch(1);
```

Comment:
You can further simplify the test by removing those latches and using futures to track task completion instead.